cmd/clusterctl/client/cluster/mover_test.go

```diff
@@ -946,6 +971,15 @@ func Test_objectMoverService_ensureNamespaces(t *testing.T) {
 			},
 			expectedNamespaces: []string{"namespace-1", "namespace-2"},
 		},
+		{
+			name: "ensureNamespaces doesn't fail if no namespace is specified (cluster-wide)",
+			fields: fields{
+				objs: externalObj.Objs(),
+			},
+			args: args{
+				toProxy: test.NewFakeProxy(),
+			},
+		},
```

Comment:
We aren't really asserting anything here. That is, if I comment out https://github.com/kubernetes-sigs/cluster-api/pull/3330/files#diff-5b99697308587ca1de3e5c4893501d8aR368-R371, all tests still pass.

Maybe we can assert that we don't get a namespace with an empty name (I didn't know this was possible TBH).
A simple check would be `g.Expect(namespaces.Items).To(HaveLen(len(tt.expectedNamespaces)))`